Feature/silabs doc (#206)
mkardous-silabs/connectedhomeip: examples/platform/silabs/README.md

# Silabs Matter Solution Guide

## Introduction

The Silabs Matter Solution Guide explains how to use the Silabs Matter offering.
It explains how to set up the development environment, build and flash a Silabs
sample app.

![Silabs logo](./images/logo.png)

> **NOTE:** Silicon Laboratories now maintains a public Matter GitHub repo with
> frequent releases thoroughly tested and validated. Developers looking to
> develop matter products with silabs hardware are encouraged to use our latest
> release with added tools and documentation.
> [Silabs Matter Github](https://github.com/SiliconLabs/matter/releases)

-   [Introduction](#introduction)
-   [Requirements](#Requirements)
    -   [Hardware Requirements](#Hardware-Requirements)
    -   [Software Requirements](#Software-Requirements)
    -   [Software Artificats](#Software-Artificats)
-   [Building](#Building)
    -   [Build Script](#Build-Script)
    -   [Build Arguments](#device-tracing)
-   [Flashing](#Flashing)

## Requirements

### Hardware Requirements

For the list of hardware requirements, see the official

> [Silicon Labs Matter HW requirements](https://siliconlabs.github.io/matter/2.2.0-1.2/general/HARDWARE_REQUIREMENTS.html)

Reviewer: they link devrait utilisé latest au lieu de la version du release example 
```suggestion
[Silicon Labs Matter HW requirements](https://siliconlabs.github.io/matter/latest/general/HARDWARE_REQUIREMENTS.html)
```

documentation.

### Software Requirements

For the list of software requirements, see the official
[Silicon Labs Matter Software requirements](https://siliconlabs.github.io/matter/2.2.0-1.2/general/SOFTWARE_REQUIREMENTS.html)documentation.

#### Software Artificats

For pre-built binaries for the latest Silicon Labs Matter release, see the
official
[Silicon Labs Matter Software Artifacts](https://siliconlabs.github.io/matter/2.2.0-1.2/general/ARTIFACTS.html#matter-software-artifacts).
This includes all necessary binairies to run a Silicon Labs sample app.

## Building

Silicon Labs currently supports the following list of sample apps in the main

> Matter SDK. Each Sample has its own documentation to explicit its specificities.

Reviewer: C'est peut-être juste moi mais je trouve ça sonne weird. Maybe this ?
```suggestion
Matter SDK. Every sample has its own documentation explaining its unique features and functionalities.
```

The examples in the `CSA Matter Repository` column are supported in the main

> Matter SDK. The examples in
> <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a> are
> supported in the Silabs Matter repository.

Reviewer: ```suggestion
Matter SDK.  Additionally, the <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a> offers extra sample applications for different device-types
```

To build the lighting app as an Wi-Fi MG24 + SiWx917 NCP, the default build command for the BRD4187C is
```shell
./scripts/examples/gn_silabs_example.sh examples/lighting-app/silabs/ out/lighting-app_siwx917 BRD4187C use_external_flash=false chip_enable_ble_rs911x=true --wifi SiWx917
```
> **Note**: The build argument  `--wifi SiWx917` is necessary to build BRD4187C image with the necessary code for the NCP combo.
> The MG24 + SiWx917 NCP combo does not yet support external flash.

To build the lighting app as an Wi-Fi MG24 + wf200 NCP, the default build command for the BRD4187C is
```shell
$ ./scripts/examples/gn_silabs_example.sh examples/lighting-app/silabs/ out/lighting-app_wf200 BRD4187C --wifi wf200
```
> **Note**: The build argument  `--wifi wf200` is necessary to build the BRD4187C image with the necessary code for the NCP combo.

#### Build Arguments

The ``gn_silabs_examples.sh` script takes two types of build arguments.
The first type are Macros processed within the script itself and the second are GN arguments.
The Macros encapsulate multiple multiple GN arguments to simplify enabling or disabled specific features.

> 

Reviewer: maybe un petit `./scripts/examples/gn_silabs_examples.sh ` will print a HELP

For the list of software requirements, see the official
[Silicon Labs Matter Software requirements](https://siliconlabs.github.io/matter/latest/general/SOFTWARE_REQUIREMENTS.html)documentation.

#### Software Artificats

For pre-built binaries for the latest Silicon Labs Matter release, see the
official
[Silicon Labs Matter Software Artifacts](https://siliconlabs.github.io/matter/latest/general/ARTIFACTS.html#matter-software-artifacts).
This includes all necessary binairies to run a Silicon Labs sample app.

## Building

Silicon Labs currently supports the following list of sample apps in the main
Matter SDK. Every sample has its own documentation explaining its unique features and functionalities.
The examples in the `CSA Matter Repository` column are supported in the main Matter SDK.
Additionally, the <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a> offers extra sample applications for different device-types

<table>
    <tbody>
        <tr>
            <th>CSA Matter Repository</th>
            <th> <a href="https://github.com/SiliconLabs/matter">Silabs Matter Repository</a></th>
        </tr>
        <tr>
          <td>
            <ul>

>                 <li> <a href="examples/lighting-app/silabs/README.md">Lighting App</a></li>

Author: all paths are wrong
